server/image_pull.go

```diff
@@ -93,6 +94,10 @@ func (s *Server) PullImage(ctx context.Context, req *types.PullImageRequest) (*t
 	}
 
 	if pullOp.err != nil {
+		if errors.Is(pullOp.err, syscall.ECONNREFUSED) {
+			// from https://github.com/kubernetes/kubernetes/blob/806b301/pkg/kubelet/images/types.go#L40-L41
+			return nil, errors.New("RegistryUnavailable")
```

Comment:
I am more in favor of updating vendor and using the variable directly. is that an option?

Reply:
We can do that, but it requires an OTEL bump since we're too far ahead of time with our dependencies. 

Ref: https://github.com/kubernetes/kubernetes/pull/117743

Reply:
Looks like we do not see an update in the near future because it introduces a huge amount of transient dependencies. Going for an override for now